v22.2/alter-index.md

```diff
@@ -5,64 +5,421 @@ toc: true
+
+You cannot `DISCARD` any zone configurations on multi-region tables, indexes, or partitions if the [multi-region abstractions](migrate-to-multiregion-sql.html#replication-zone-patterns-and-multi-region-sql-abstractions) created the zone configuration.
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER INDEX vehicles@vehicles_auto_index_fk_city_ref_users CONFIGURE ZONE DISCARD;
+~~~
+
+### Define partitions
+
+#### Define a list partition on an index
+
+Suppose we have a table called `students_by_list`, a secondary index on the table called `name_idx`, and the primary key of the table is defined as `(country, id)`. We can define partitions on the index by list:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER INDEX students_by_list@name_idx PARTITION BY LIST (country) (
+    PARTITION north_america VALUES IN ('CA','US'),
+    PARTITION australia VALUES IN ('AU','NZ'),
+    PARTITION DEFAULT VALUES IN (default)
+  );
+~~~
+
+#### Define a range partition on an index
+
+Suppose we have a table called `students_by_range`, a with a secondary index called `name_idx`, and the primary key of the table is defined as `(expected_graduation_date, id)`. We can define partitions on the index by range:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER INDEX students_by_range@name_idx PARTITION BY RANGE (expected_graduation_date) (
+    PARTITION graduated VALUES FROM (MINVALUE) TO ('2017-08-15'),
+    PARTITION current VALUES FROM ('2017-08-15') TO (MAXVALUE)
+  );
+~~~
+
+#### Define subpartitions on an index
+
+Suppose we have a table named `students`, with a secondary index called `name_idx`, and the primary key is defined as `(country, expected_graduation_date, id)`. We can define partitions and subpartitions on the index:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER INDEX students@name_idx PARTITION BY LIST (country) (
+    PARTITION australia VALUES IN ('AU','NZ') PARTITION BY RANGE (expected_graduation_date) (
+      PARTITION graduated_au VALUES FROM (MINVALUE) TO ('2017-08-15'),
+      PARTITION current_au VALUES FROM ('2017-08-15') TO (MAXVALUE)
+    ),
+    PARTITION north_america VALUES IN ('US','CA') PARTITION BY RANGE (expected_graduation_date) (
+      PARTITION graduated_us VALUES FROM (MINVALUE) TO ('2017-08-15'),
+      PARTITION current_us VALUES FROM ('2017-08-15') TO (MAXVALUE)
+    )
+  );
+~~~
+
+#### Repartition an index
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER INDEX students_by_range@name_idx PARTITION BY RANGE (expected_graduation_date) (
+    PARTITION graduated VALUES FROM (MINVALUE) TO ('2018-08-15'),
+    PARTITION current VALUES FROM ('2018-08-15') TO (MAXVALUE)
+  );
+~~~
+
+#### Unpartition an index
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER INDEX students@name_idx PARTITION BY NOTHING;
+~~~
+
+### Rename indexes
+
+#### Rename an index
```

Comment:
Unfortunately yeah, to be consistent with how these example headings are named, and to avoid breaking any pre-existing links to the specific named example (short of manually changing them all), I need to keep both h4 and h5 here. I'll think on ways to improve this, however.